### DISCUSSION TITLE
AlterInstaller and using avbroot to switch between rooted and unrooted OTAs

### DISCUSSION THREAD

**Discussion by MvRoo** [chenxiaolong/avbroot#351] [Q&A] [answered]
Hi,

I have been going through a couple of your repos in the last few days, after I'd initially found AlterInstaller. Your avbroot and my-avbroot-setup repos have really intrigued me; I have been running custom roms since forever and have been switching back and forward between CalyxOS and GrapheneOS for the last year, because I want a privacy and security centric ROM (without Google crap), It wasn't until I found your repos that I realized that rooting would potentially also be possible, while remaining secure.

Now, because it's been a very long time since I have run a rooted ROM, I am not all that up-to-date anymore, so forgive me for the potentially stupid question:

I read another discussion where it was mentioned that it is possible to switch between rooted and unrooted OTAs: #313. That sounds great!

Would it be possible to install an app like OsmAnd through F-droid, then flash a rooted OTA to use AlterInstaller to patch/spoof the installer fields for it, and after that go back to a --rootless version of the OTA? Or would I be patching something in system and would then run into the problem mentioned in #339 ?

Thanks in advance for taking the time (if so) and big kudos for all your open-source work!

**Comment by chenxiaolong** [comment, marked as answer]
Glad you like my projects!

> Would it be possible to install an app like OsmAnd through F-droid, then flash a rooted OTA to use AlterInstaller to patch/spoof the installer fields for it, and after that go back to a --rootless version of the OTA?

The file modified by AlterInstaller is `/data/system/packages.xml`, which is just a regular data file that stores all of Android's metadata about all the installed apps. You can do exactly what you suggested and it should work fine.

**Comment by pixincreate** [comment]
I might be a selfish / biased here, but hey, you can try out my new tool called as [`PixeneOS`](https://github.com/pixincreate/PixeneOS) which is a mixture of all @chenxiaolong's projects.

You just have to do the initial set up (of setting up the signing keys) and add your device name in the `env.toml`, you should be good to go.  

I never got the time to actually update the documentation lately, but yeah, it works now.  
I haven't personally tested this as the pixel that i own is my daily driver, but planning to shift to it eventually.

With PixeneOS, you just have to change the OTA URL to point to rooted OTA, and you should be getting an update in Custota (theoritically w.r.t to what I've done). But yeah, you'll have to build rooted OTA manually which takes about 7 - 8 mins.

**Reply by MvRoo** [reply]
Thanks for making me aware! I might give it a go once I've done it once successfully using @chenxiaolong's guide.
